exercises/go-1-a/go_1_a_test.go

```diff
@@ -0,0 +1,42 @@
+package go1a
+
+import "testing"
+
+func TestTwelve(t *testing.T) {
+	AssertDescribe(t, 12, "1 dozen or 12")
+}
+
+func TestTwenty(t *testing.T) {
+	AssertDescribe(t, 20, "1 score or 20")
+}
+
+func TestOneHundredFourtyFour(t *testing.T) {
+	AssertDescribe(t, 144, "1 gross or 12 dozen or 144")
+}
+
+func TestOne(t *testing.T) {
+	AssertDescribe(t, 1, "1")
+}
+
+func TestFourty(t *testing.T) {
+	AssertDescribe(t, 40, "2 score or 40")
+}
+
+func TestSeventyTwo(t *testing.T) {
+	AssertDescribe(t, 72, "6 dozen or 72")
+}
+
+func TestTwoHundredEightyEight(t *testing.T) {
+	AssertDescribe(t, 288, "2 gross or 24 dozen or 288")
+}
+
+func TestSixtyFive(t *testing.T) {
+	AssertDescribe(t, 65, "65")
+}
+
+func AssertDescribe(t *testing.T, input int, expected string) {
```

Comment:
Table driven testing, instead of single functions?

Reply:
Agree. Would be better.

Reply:
Sample of a table driven test:

```go
func TestDescribe(t *testing.T) {
	type args struct {
		amount int
	}
	tests := []struct {
		name string
		args args
		want string
	}{
		{
			name: "Twelve",
			args: args{
				amount: 12,
			},
			want: "1 dozen or 12",
		},
		{
			name: "Twenty",
			args: args{
				amount: 20,
			},
			want: "1 score or 20",
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := Describe(tt.args.amount); got != tt.want {
				t.Errorf("Describe(%d) = %q, expected %q.", tt.args.amount, got, tt.want)
			}
		})
	}
}
```

The tests slice could also be taken out of the function as it is done with most exercism tests. I prefer the tests in the function to avoid naming collisions.

Reply:
This looks nice, but this has one unfortunate side-effect: test runner also includes the parent tests in the test results:

```json
{
	"status": "fail",
	"tests": [
		{
			"name": "TestDescribe",
			"status": "fail",
			"expected": "",
			"actual": "",
			"message": "\n=== RUN   TestDescribe\n\n--- FAIL: TestDescribe (0.00s)\n"
		},
		{
			"name": "TestDescribe/Twelve",
			"status": "pass",
			"expected": "",
			"actual": "",
			"message": "\n=== RUN   TestDescribe/Twelve\n\n    --- PASS: TestDescribe/Twelve (0.00s)\n"
		},
		{
			"name": "TestDescribe/Twenty",
			"status": "fail",
			"expected": "",
			"actual": "",
			"message": "\n=== RUN   TestDescribe/Twenty\n\n    --- FAIL: TestDescribe/Twenty (0.00s)\n\n        go_1_a_test.go:32: Describe(20) = \"20\", expected \"1 score or 20\".\n"
		}
	]
}
```

This would mean that the GUI would show `TestDescribe` failing, and will not know for which input the test failed. If we really want the table-driven testing, the test runner should remove this top-level tests from the test results. 

I'd prefer to get this merged and then afterwards update the test suite when the test runner has been updated.

Reply:
An [issue](https://github.com/exercism/go-test-runner/issues/8) has been created to update the test runner.